tests/unittests/RecommendationSystemTest.cpp

```diff
@@ -892,10 +908,21 @@ class RecommendationSystemTest : public BackendTest {
     // Use the same precision transformation for compilation.
     CompilationContext cctx;
     cctx.precisionConfig = precConfig_;
-    EE_.compile(cctx);
+    std::vector<std::unique_ptr<DeviceConfig>> configs;
```

Comment:
The new code here seems to have a lot of boiler-plate similar to some other parts you introduced. Could you try to move this common logic into a helper function?